templates/python/custom-copilot-basic/src/bot.py.tpl

```diff
@@ -65,4 +62,9 @@ async def on_error(context: TurnContext, error: Exception):
     traceback.print_exc()
 
     # Send a message to the user
-    await context.send_activity("The bot encountered an error or bug.")
\ No newline at end of file
+    await context.send_activity("The bot encountered an error or bug.")
+
+@bot_app.feedback_loop()
+async def feedback_loop(_context: TurnContext, _state: TurnState, feedback_loop_data: FeedbackLoopData):
+    # Add custom feedback process logic here.
+    print(f"Your feedback is:\n{json.dumps(json.loads(feedback_loop_data.to_json()), indent=4)}")
```

Comment:
Are you trying to prettify the format? Serialize the object, deserialize then serialize again makes the logic a little wired. Not sure whether we can just print `feedback_loop_data.to_json()`

Reply:
`json.dumps(asdict(feedback_loop_data), indent=4)`, a possible way to make it neat. FYI.

Reply:
that makes sense and works.